src/Admin/AbstractAdmin.php

```diff
@@ -2487,6 +2500,24 @@ public function toString($object)
         return sprintf('%s:%s', ClassUtils::getClass($object), spl_object_hash($object));
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function subjectAsString(): string
+    {
+        if (!$this->hasSubject()) {
+            throw new \LogicException(sprintf('Subject is not set for admin "%s".', $this->getCode()));
+        }
+
+        $subject = $this->getSubject();
+
+        if (\is_callable([$subject, '__toString']) && null !== $toString = $subject->__toString()) {
+            return $toString;
+        }
+
+        return sprintf('%s:%s', ClassUtils::getClass($subject), spl_object_hash($subject));
```

Comment:
Can we use the primary key for the object here?

Reply:
I think that could be possible only for model managers that use and expose that feature.